src/script/components/asset/controls/audioSeekBar.js

```diff
@@ -17,114 +17,105 @@
-    component_info.element.addEventListener('click', this._on_level_click);
-    window.addEventListener('resize', this._on_resize_fired);
+    this._onLevelClick = this._onLevelClick.bind(this);
+    this._onTimeUpdate = this._onTimeUpdate.bind(this);
+    this._onAudioEnded = this._onAudioEnded.bind(this);
+    this.audioElement.addEventListener('ended', this._onAudioEnded);
+    this.audioElement.addEventListener('timeupdate', this._onTimeUpdate);
+    this.element.addEventListener('click', this._onLevelClick);
+    window.addEventListener('resize', this._onResizeFired);
   }
 
-  _render_levels() {
-    const number_of_levels_fit_on_screen = Math.floor(this.element.clientWidth / 3); // 2px + 1px
-    const scaled_loudness = z.util.ArrayUtil.interpolate(this.loudness, number_of_levels_fit_on_screen);
+  _renderLevels() {
+    const numberOfLevelsFitOnScreen = Math.floor(this.element.clientWidth / 3); // 2px + 1px
+    const scaledLoudness = ArrayUtil.interpolate(this.loudness, numberOfLevelsFitOnScreen);
+    this.element.innerHTML = '';
 
-    // eslint-disable-next-line no-unsanitized/property
-    this.element.innerHTML = scaled_loudness.map(level => `<span style="height: ${level}px"></span>`).join('');
+    this.levels = scaledLoudness.map(loudness => {
+      const level = document.createElement('span');
+      level.style.height = `${loudness}px`;
+      this.element.appendChild(level);
+      return level;
+    });
   }
 
-  _normalize_loudness(loudness, max) {
+  _normalizeLoudness(loudness, max) {
     const peak = Math.max(...loudness);
     const scale = max / peak;
     return peak > max ? loudness.map(level => level * scale) : loudness;
   }
 
-  _on_level_click(event) {
-    const mouse_x = event.pageX - $(event.currentTarget).offset().left;
-    const calculatedTime = (this.audio_element.duration * mouse_x) / event.currentTarget.clientWidth;
+  _onLevelClick(event) {
+    const mouse_x = event.pageX - event.currentTarget.getBoundingClientRect().left;
+    const calculatedTime = (this.audioElement.duration * mouse_x) / event.currentTarget.clientWidth;
     const currentTime = window.isNaN(calculatedTime) ? 0 : calculatedTime;
 
-    this.audio_element.currentTime = z.util.NumberUtil.clamp(currentTime, 0, this.audio_element.duration);
-    this._on_time_update();
-  }
-
-  _on_time_update() {
-    const $levels = this._clear_theme();
-    const index = Math.floor((this.audio_element.currentTime / this.audio_element.duration) * $levels.length);
-    this._add_theme(index);
-  }
-
-  _on_audio_ended() {
-    this._clear_theme();
+    this.audioElement.currentTime = NumberUtil.clamp(currentTime, 0, this.audioElement.duration);
+    this._onTimeUpdate();
   }
 
-  _clear_theme() {
-    return $(this.element)
-      .children()
-      .removeClass('active');
+  _onTimeUpdate() {
+    const index = Math.floor((this.audioElement.currentTime / this.audioElement.duration) * this.levels.length);
```

Comment:
Would be nice to consume `event.target` instead of `this.audioElement`) to make the function less bound to context